### PR TITLE
Org event sharing

### LINKED ISSUE #170: Feat: Sharing an Org/Event
We want users to be able to share specfic orgs and evnts with their friends. When clicking on the share button from the org and event pages this popup should show

Image

Shared should create a link that will route directly to the page they share

### PR DESCRIPTION
**Pull request by KarimJC**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/170)
Please include a summary of the changes and the related issue. Please also
include relevant motivation, context, and images!

This ticket adds sharing capabilities to organizations and events through a modal. Hard to test that the links actually work though because this is still local.

# How Has This Been Tested?
Please describe the tests that you manually ran to verify your changes (beyond any unit/integration tests written and ran).

## Screenshots
Please provide screenshots of manual testing (Scalar, frontend pages, etc.)

# Checklist

## General
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]
image
image

### Android
[Add Android screenshots here]

### CODE REVIEW

File: frontend/apps/mobile/app/(app)/event/[id].tsx [hunk 1]
```diff
@@ -8,21 +8,24 @@ import {
...
 } from "react-native";
+import * as Linking from "expo-linking";
 import { SafeAreaView } from "react-native-safe-area-context";
 import { useLocalSearchParams, useRouter } from "expo-router";
 import {
   useGetEventOccurrencesByEventId,
   useGetReviewAggregate,
 } from "@skillspark/api-client";
 import type { EventOccurrence } from "@skillspark/api-client";
+import MaterialIcons from "@expo/vector-icons/MaterialIcons";
```
Comment (adescoteaux1) [inline frontend/apps/mobile/app/(app)/event/[id].tsx]: can we use icon symbol instead of material icons

File: frontend/apps/mobile/app/(app)/org/[id].tsx [hunk 2]
```diff
@@ -95,36 +99,46 @@ function OrgDetail({
...
+                  {location.district}, {location.province}
+                </Text>
+              )}
+              <View className="flex-row items-center gap-1.5">
+                <Text className="text-[14px]">🔥</Text>
+                <Text
+                  className="text-[14px] font-nunito"
+                  style={{ color: AppColors.mutedText }}
+                >
+                  100+ {translate("org.bookingsThisWeek")}
```
Comment (adescoteaux1) [inline frontend/apps/mobile/app/(app)/org/[id].tsx]: why did we bring back hard coding

File: frontend/apps/mobile/components/ShareModal.tsx [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+      onPress: () => openAppOrStore(
+        `whatsapp://send?text=${encodeURIComponent(text)}`,
+        "whatsapp"
+      ),
+      render: () => <FontAwesome5 name="whatsapp" size={20} color="#fff" />,
+    },
+  ];
+}
+
+function CircleIcon({
```
Comment (adescoteaux1) [inline frontend/apps/mobile/components/ShareModal.tsx]: move component to separate file

**Review by adescoteaux1 — COMMENTED** [review]
couple things on an initial pass

**Comment by KarimJC** [comment]
@adescoteaux1 it is ready

**Comment by KarimJC** [comment]
@adescoteaux1 ready again

File: frontend/apps/mobile/components/ShareModal.tsx [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+export function ShareModal({
+  visible,
+  onClose,
+  name,
+  imageUrl,
+  shareUrl,
+  message,
+}: ShareModalProps) {
+  const insets = useSafeAreaInsets();
+  const { t } = useTranslation();
```
Comment (josh-torre) [inline frontend/apps/mobile/components/ShareModal.tsx]: make an alias to translate

File: frontend/apps/mobile/components/AboutPage.tsx [hunk 5]
```diff
@@ -10,13 +11,12 @@ interface AboutPageProps {
 
 export function AboutPage({ description, links }: AboutPageProps) {
   const { openLink, hasLinks } = useOrgLinks(links);
+  const { t } = useTranslation();
```
Comment (josh-torre) [inline frontend/apps/mobile/components/AboutPage.tsx]: Alias to translate

**Review by josh-torre — COMMENTED** [review]
Great work overall :)

File: frontend/apps/mobile/app/(app)/event/[id].tsx (hunk 1 above)
Comment (josh-torre) [reply frontend/apps/mobile/app/(app)/event/[id].tsx]: @KarimJC Please also make sure you switch to using our icons here:  per @adescoteaux1 's msg

**Comment by KarimJC** [comment]
@josh-torre working on merge conflicts rn

**Comment by KarimJC** [comment]
@josh-torre @adescoteaux1 all done

File: frontend/apps/mobile/app/(app)/(tabs)/index.tsx [hunk 6]
```diff
@@ -325,26 +329,27 @@ export default function HomeScreen() {
...
-              </View>
-            ))}
+        {/* Explore by Category */}
+        {categories.length > 0 && (
+          <View className="mb-6">
+            <Text
+              className="font-nunito-bold px-5 mb-3"
+              style={{ fontSize: FontSizes.lg, color: AppColors.primaryText }}
+            >
+              Explore by Category
```
Comment (adescoteaux1) [inline frontend/apps/mobile/app/(app)/(tabs)/index.tsx]: Noticed that the text is not all translatable on this file, can you fix it, sorry

File: frontend/apps/mobile/app/(app)/event/[id].tsx (hunk 1 above)
Comment (adescoteaux1) [reply frontend/apps/mobile/app/(app)/event/[id].tsx]: bumping this - icon symbol > material icons

File: frontend/apps/mobile/components/AboutPage.tsx [hunk 7]
```diff
@@ -10,13 +11,12 @@ interface AboutPageProps {
 
 export function AboutPage({ description, links }: AboutPageProps) {
   const { openLink, hasLinks } = useOrgLinks(links);
+  const { t: translate } = useTranslation();
   const [expanded, setExpanded] = useState(false);
   const [truncated, setTruncated] = useState(false);
 
   return (
     <View>
-      <Text className="text-lg font-bold text-gray-900 mb-2.5">About</Text>
```
Comment (adescoteaux1) [inline frontend/apps/mobile/components/AboutPage.tsx:18]: why delete?

File: frontend/apps/mobile/components/ShareModal.tsx [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+        const smsUrl =
+          Platform.OS === "ios"
+            ? `sms:&body=${encodeURIComponent(text)}`
+            : `sms:?body=${encodeURIComponent(text)}`;
+        Linking.openURL(smsUrl).catch(() => {});
+      },
+      render: () => <IconSymbol name="message.fill" size={22} color="#fff" />,
+    },
+    {
+      name: "Messenger",
```
Comment (adescoteaux1) [inline frontend/apps/mobile/components/ShareModal.tsx]: translate these texts

File: frontend/apps/mobile/utils/sharing.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import { Share } from "react-native";
+import * as Linking from "expo-linking";
+
+export async function shareEventLink(id: string, title?: string) {
+  const url = Linking.createURL(`event/${id}`);
+  await Share.share({
+    message: title ? `Check out "${title}" on SkillSpark: ${url}` : url,
```
Comment (adescoteaux1) [inline frontend/apps/mobile/utils/sharing.ts:7]: should we send this text in the users language maybe?

**Review by adescoteaux1 — COMMENTED** [review]
did you end up being able to test?

File: frontend/apps/mobile/utils/sharing.ts (hunk 9 above)
Comment (KarimJC) [reply frontend/apps/mobile/utils/sharing.ts:7]: probably

**Review by adescoteaux1 — APPROVED** [review]
lgtm
